lib/client-assets.php

```diff
@@ -1002,9 +1047,11 @@ function gutenberg_editor_scripts_and_styles( $hook ) {
 	$script .= sprintf( 'var editorSettings = %s;', wp_json_encode( $editor_settings ) );
 	$script .= <<<JS
 		window._wpLoadGutenbergEditor = new Promise( function( resolve ) {
-			wp.api.init().then( resolve );
-		} ).then( function() {
-			return wp.editPost.initializeEditor( 'editor', window._wpGutenbergPost, editorSettings );
+			wp.api.init().then( function() {
+				wp.domReady.default( function() {
```

Comment:
I think it's the issue with how Webpack is configured. `wp.domReady` is an object with the current setup:

![screen shot 2018-05-11 at 09 02 08](https://user-images.githubusercontent.com/699132/39911011-1795cbbc-54fa-11e8-9d93-5a47b3446b1e.png)

Reply:
I dug into this a bit, and it's trickier than it appears on the surface.

This is what [Webpack's `output.libraryExport` configuration](https://webpack.js.org/configuration/output/#output-libraryexport) is meant to support: Setting a specific property as the value assigned on the library target (`window`).

However, we have mixed needs: For something like `wp.editor` or `wp.i18n`, we want named exports to be respected, and assigned as nested properties within. For `dom-ready` though, the default export function should be assigned as `wp.domReady`. 

In my head, it's as though we want `libraryExport: 'default'` when there are no other named exports, otherwise using the default `_entry_return_`. 

So far as implementation, we might need to get our hands dirty with a custom Webpack plugin to achieve this. It probably needs to extend (whether via classical inheritance or just dynamically instantiating during `apply`, I'm uncertain) Webpack's [`ExportPropertyMainTemplatePlugin.js`](https://github.com/webpack/webpack/blob/master/lib/ExportPropertyMainTemplatePlugin.js). Where I started having trouble was determining the exports of a module to decide what to assign as `libraryExport`.

Reply:
Alternative: have separate `output` configurations for specific edge-case modules like `dom-ready`, where we can assign `libraryExport: 'default'`. This too is not made very simple in an all-in-one Webpack configuration like we have.

Alternative alternative: Don't allow `default` exports on modules.

Alternative alternative alternative: Use CommonJS `module.exports` instead of `export default` from the `packages` implementation.

Reply:
Alternative Alternative Alternative, assign the globals manually in "enqueues" importing things from the regular built modules :)

Reply:
Moving task to #6748 for further discussion.